guide/src/mock_a_client.md

```diff
@@ -101,6 +101,18 @@ Finally, we are ready to call our function...
 {{#include ../samples/tests/mocking.rs:validate}}
 ```
 
+### Simulating errors
+
+Simulating errors is no different than simulating successes. We just need to
+modify the result returned by our mock.
+
+```rust,ignore
+{{#include ../samples/tests/mocking.rs:error}}
+```
+
+Note that a client built `from_stub()` does not have an internal retry loop. It
+returns all errors from the stub directly to the application.
+
```

Comment:
In other sections I link the full function for each sub-section separately. Consider doing the same here.

Reply:
> Um, I think I will pass. The tests have shared code, so I do not think the sub-sections would actually be "full".

Strike this. I just realized we can have an anchor in multiple places, and `mdbook` will concatenate the sections. e.g.

```
// ANCHOR: foo
1
// ANCHOR_END: foo
2
3
4
// ANCHOR: foo
5
// ANCHOR_END: foo
```

yields:

```
1
5
```

So I am out of excuses. Will clean this up soon-ish.

Reply:
Nevermind. The `mdbook` behavior I described is totally unreliable. :(

Reply:
> totally unreliable

Lol, I finally understand...

`{{#include ... }}`  will not concatenate sections with the same marker.

`{{#rustdoc_include ... }}` will concatenate sections with the same marker.

I was switching between the two unreliably.